### ISSUE TITLE
wg can't start because iptables-restore fails

### ISSUE THREAD

**Issue by Unlifate** [linuxserver/docker-wireguard#156]
[linuxserverurl]: https://linuxserver.io
[![linuxserver.io](https://raw.githubusercontent.com/linuxserver/docker-templates/master/linuxserver.io/img/linuxserver_medium.png)][linuxserverurl]

------------------------------

## Expected Behavior

Starting the container should bring up a working connection. 

## Current Behavior

The wireguard startup within the container fails:
```
[#] iptables-restore -n
iptables-restore: line 7 failed
```
It would help me a lot if someone could tell me how to log the lines that `iptables -n` fails.

## Steps to Reproduce

1. Have my environment (see below)
2. Create a [`docker-compose.yml`](https://github.com/linuxserver/docker-wireguard/files/8340842/docker-compose.txt)
3. Provide a working wg0 config file
4. Compose and run container in detached mode `docker compose up -d`
5. Check log `docker logs wireguard`

## Environment
**OS:** Windows 10 Build 19044 with WSL2
**CPU architecture:** x86_64
**How docker service was installed:** Docker Desktop 4.6.1 Installer

## Command used to create docker container (run/create/compose/screenshot)

See above

## Docker logs

[docker.log](https://github.com/linuxserver/docker-wireguard/files/8340834/docker.log)

**Comment by benjick** [comment]
Seems to be Windows specific. It worked fine on my Mac when I was developing but when I pulled to run it on my Windows machine I got the same error as you have

**Comment by Unlifate** [comment]
It is furthermore Windows WSL2 specific.
For a temporary fix: disabled WSL2 and enabled Hyper-V.

**Comment by benjick** [comment]
Any suggestions that does not revolve around disabling WSL?

**Comment by henryclw** [comment]
> Any suggestions that does not revolve around disabling WSL?

Perhaps you would find these answers useful:

- <https://unix.stackexchange.com/questions/700620/run-wireguard-as-a-client-on-win10-with-wsl2/700621>
- <https://www.reddit.com/r/bashonubuntuonwindows/comments/jk4x24/is_there_a_way_to_run_wireguard_within_wsl2/gah860j>

But compiling the kernel seems to be a bit hard for me, I don't have many experience with this.

**Comment by benjick** [comment]
Why close?

**Comment by aptalca** [comment]
Because it's not a container issue. It's an environment issue specific to windows, as you stated above.

**Comment by henryclw** [comment]
@benjick 

Hi, I've found a solution for VPN, but not using wireguard, it's using tailscale. I don't know how but the container worked and all the traffic is routed to an exit node in tailscale. I tried to check their dockerfile to understand how this could be done, but I didn't find anything special.

- <https://github.com/tailscale/tailscale/blob/main/Dockerfile>

Maybe this could be a hint or so, hope this could be you out.
